src/Compiler/Driver/CompilerOptions.fs

```diff
@@ -1972,31 +1972,31 @@ let deprecatedFlagsFsc tcConfigB =
 // OptionBlock: Miscellaneous options
 //-----------------------------------
 
-let DisplayBannerText tcConfigB =
+let DisplayBannerText tcConfigB print =
     if tcConfigB.showBanner then
-        (printfn "%s" tcConfigB.productNameForBannerText
-         printfn "%s" (FSComp.SR.optsCopyright ()))
+        (print $"{tcConfigB.productNameForBannerText}\n"
+         print $"{FSComp.SR.optsCopyright ()}\n")
 
 /// FSC only help. (FSI has it's own help function).
-let displayHelpFsc tcConfigB (blocks: CompilerOptionBlock list) =
-    DisplayBannerText tcConfigB
-    PrintCompilerOptionBlocks blocks
-    exit 0
+let DisplayHelpFsc tcConfigB (blocks: CompilerOptionBlock list) print exit =
+    DisplayBannerText tcConfigB print
+    PrintCompilerOptionBlocks blocks print
+    exit()
 
-let displayVersion tcConfigB =
-    printfn "%s" tcConfigB.productNameForBannerText
-    exit 0
+let DisplayVersion tcConfigB print exit =
+    print $"{tcConfigB.productNameForBannerText}\n"
+    exit()
 
 let miscFlagsBoth tcConfigB =
     [
         CompilerOption("nologo", tagNone, OptionUnit(fun () -> tcConfigB.showBanner <- false), None, Some(FSComp.SR.optsNologo ()))
-        CompilerOption("version", tagNone, OptionConsoleOnly(fun _ -> displayVersion tcConfigB), None, Some(FSComp.SR.optsVersion ()))
+        CompilerOption("version", tagNone, OptionConsoleOnly(fun _ -> DisplayVersion tcConfigB Console.Write (exit 0)), None, Some(FSComp.SR.optsVersion ()))
```

Comment:
(exit 0) here will probably evaluate, what you want is to pass a closure.

Reply:
Yes, fixed